ninja.lua

```diff
@@ -251,8 +251,15 @@ local function getcflags(toolset, cfg, filecfg)
 	local cppflags = ninja.list(toolset.getcppflags(filecfg))
 	local cflags = ninja.list(toolset.getcflags(filecfg))
 	local defines = ninja.list(table.join(toolset.getdefines(filecfg.defines), toolset.getundefines(filecfg.undefines)))
-	local includes = ninja.list(toolset.getincludedirs(cfg, filecfg.includedirs, filecfg.externalincludedirs, filecfg.frameworkdirs, filecfg.includedirsafter))
-	local forceincludes = ninja.list(toolset.getforceincludes(cfg))
+	
+	-- ninja expects include directories to be relative to the root folder.
+	-- what follows is a little hack to trick the toolsets into thinking cfg.workspace is actually cfg.project
+	-- and will therefore use that as the base for the relative paths
+	-- this is an issue with premake itself.
+	local cfg_override = cfg
+	cfg_override.project = cfg.workspace
+	local includes = ninja.list(toolset.getincludedirs(cfg_override, filecfg.includedirs, filecfg.externalincludedirs, filecfg.frameworkdirs, filecfg.includedirsafter))
+	local forceincludes = ninja.list(toolset.getforceincludes(cfg_override))
```

Comment:
That hack is superfluous with the other PR which handles it correctly (even if it requires change from premake-core)